src/Microsoft.AspNetCore.Mvc.Core/IStatusCodeActionResult.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc
```

Comment:
Let's put this in `Microsoft.AspNetCore.Mvc.Infrastructure`. It's not an interface that we'd expect users to typically interact with.